libnavui-util/src/main/java/com/mapbox/navigation/ui/utils/internal/lifecycle/ViewLifecycleOwner.kt

```diff
@@ -0,0 +1,89 @@
+package com.mapbox.navigation.ui.utils.internal.lifecycle
+
+import android.view.View
+import androidx.annotation.UiThread
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.LifecycleRegistry
+import androidx.lifecycle.lifecycleScope
+import androidx.lifecycle.repeatOnLifecycle
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import java.util.concurrent.Flow
+
+/**
+ * A [LifecycleOwner] that merges the [hostingLifecycleOwner]'s [Lifecycle] (of an Activity or Fragment)
+ * with the [View]'s drawing surface lifecycle.
+ *
+ * This lifecycle is useful for continuously running tasks that obtain the data and draw it in the [View],
+ * for example by collecting [Flow]s and drawing results in the nested views or on the map.
+ *
+ * In general, this lifecycle reflects the [hostingLifecycleOwner]'s [Lifecycle] whenever the `View` is attached
+ * but whenever the `View` is detached and we want to stop the coroutines tied to the lifecycle's scope,
+ * this lifecycle will transition to [Lifecycle.State.CREATED] (and invoke [Lifecycle.Event.ON_STOP])
+ * even if the hosting Activity or Fragment is still running.
+ *
+ * This lifecycle reaches [Lifecycle.State.DESTROYED] only when the [hostingLifecycleOwner] is destroyed.
+ *
+ * @see keepExecutingWhenStarted
+ */
+@UiThread
+class ViewLifecycleOwner(
+    private val hostingLifecycleOwner: LifecycleOwner
```

Comment:
On the topic of orientation changes. We could also add the ability to handle the orientation change in here

https://github.com/mapbox/mapbox-navigation-android/blob/f337d446971c5fbd4cce29408173aaa31c8d5640/libnavigation-core/src/main/java/com/mapbox/navigation/core/lifecycle/CarAppLifecycleOwner.kt#L85-L87

Reply:
Could you elaborate why would we need it here? Wondering what should be hooked up to a configuration check.

Reply:
I agree that we should first have a reason to hook it up. I was "thinking out loud" with this comment.

Right now our solution for handling configuration changes is to put state in a ViewModel or hook it to the `MapboxNavigationApp.lifecycle`. If we were handling orientation changes ourselves, maybe we don't really need to use `ViewModel`.

But resolving this, conversation is out of scope